js/plugins/googleai/src/index.ts

```diff
@@ -173,15 +178,17 @@ async function resolver(
   actionName: string,
   options?: PluginOptions
 ) {
-  switch (actionType) {
-    case 'model':
-      resolveModel(ai, actionName, options);
-      break;
-    case 'embedder':
-      resolveEmbedder(ai, actionName, options);
-      break;
-    default:
-    // no-op
+  if (actionType === 'embedder') {
+    resolveEmbedder(ai, actionName, options);
+  } else if (actionName.startsWith('veo')) {
+    // we do it this way because the request may come in for
+    // action type 'model' and action name 'veo-...'. That case should
+    // be a noop. It's just the order or model lookup.
+    if (actionType === 'background-model') {
```

Comment:
It might be nicer to just have 3 switch cases 'model', 'background-model' and 'embedder' and each goes to its respective resolve\<X\>. Then in those resolveX functions you can look at the actionName and decide what you want to do with it if anything. (e.g. resolveModel does nothing with Veo, and resolveBackgroundModel does nothing without Veo ... for now... until there are more background models). This is more expandable in future without becoming a complicated page of nested conditions.